### PR TITLE
Added task-reliability.

### PR DESCRIPTION
**Pull request by jimoleary**
Added task-stats.
changed get_api to use api_server_host (for local testing).
reordered CONFIG_FILE_LOCATIONS so that ~/cli_bin is used first.

### CODE REVIEW

File: src/evergreen/api.py [hunk 1]
```diff
@@ -321,6 +323,107 @@ def test_stats_by_project(self,
...
+        :param group_num_days: Aggregate statistics to this size.
+        :param requesters: Filter by requesters (mainline, patch, trigger, or adhoc).
+        :param tasks: Only include specified tasks.
+        :param variants: Only include specified variants.
+        :param distros: Only include specified distros.
+        :param group_by: How to group results (test_task_variant, test_task, or test)
+        :param sort: How to sort results (earliest or latest).
+        :return: Patch queried for.
+        """
+        if after_date is None:
```
Comment (jimoleary) [inline src/evergreen/api.py]: different to other calls, but added defaults for dates.

File: src/evergreen/api.py [hunk 2]
```diff
@@ -514,15 +617,18 @@ def get_api(cls, auth=None, use_config_file=False, config_file=None,
...
         if not auth and use_config_file:
             config = read_evergreen_config()
             auth = get_auth_from_config(config)
+            if auth:
+                kwargs['auth'] = auth
 
-        if not auth and config_file:
-            config = read_evergreen_from_file(config_file)
-            auth = get_auth_from_config(config)
+            # If there is a value for api_server_host, then use it.
```
Comment (jimoleary) [inline src/evergreen/api.py]: without this code it always connect to real evergreen.

File: src/evergreen/api.py [hunk 3]
```diff
@@ -321,6 +323,107 @@ def test_stats_by_project(self,
...
+        :param tasks: Only include specified tasks.
+        :param variants: Only include specified variants.
+        :param distros: Only include specified distros.
+        :param group_by: How to group results (test_task_variant, test_task, or test)
+        :param sort: How to sort results (earliest or latest).
+        :return: Patch queried for.
+        """
+        if after_date is None:
+            date = datetime.utcnow() - timedelta(days=180)
+            after_date = date.strftime('%Y-%m-%d')
```
Comment (dbradf) [inline src/evergreen/api.py]: In `src/evergreen/util.py`, there is [code](https://github.com/evergreen-ci/evergreen.py/blob/32c3a9c5ac42e32f6b7c6ee816fcad5f043def7b/src/evergreen/util.py#L21-L28) to handle dates and datetimes. Can we use that so we will only have to worrry about how evergreen formats times and dates?

**Comment by dbradf** [comment]
Could you also update the version in setup.py and add your change to CHANGELOG.md.

**Pull request merged**